build.gradle

```diff
@@ -55,7 +52,17 @@ jar {
 task createJextractImage(type: Exec) {
     dependsOn jar
 
-    onlyIf { !new File("$buildDir/jextract-jdk-image").exists() }
+    // if these inputs or outputs change, gradle will rerun the task
+    inputs.file(jar.archiveFile.get())
+    outputs.dir(jextract_app_dir)
+
+    def quote_jlink_opts = Os.isFamily(Os.FAMILY_WINDOWS)?
+        '\\"--enable-native-access=org.openjdk.jextract\\"' :
+        '"--enable-native-access=org.openjdk.jextract"'
+
+    doFirst {
+        delete(jextract_app_dir)
+    }
```

Comment:
I didn't know about inputs/outputs of Exec task. I just checked that running verify without clean in between does not rebuild (after your change). So looks fine.